iso15118/secc/states/iso15118_2_states.py

```diff
@@ -791,24 +837,57 @@ async def process_message(
             # TODO Either an MO Root certificate or a V2G Root certificate
             #      could be used to verify, need to be flexible with regards
             #      to the PKI that is used.
-            verify_certs(leaf_cert, sub_ca_certs, CertPath.MO_ROOT_DER)
-
-            # TODO Check if EMAID has correct syntax
+            verify_certs(
+                leaf_cert, sub_ca_certs, self._mobility_operator_root_cert_path()
+            )
 
+            # TODO Check if EMAID has correct syntax -- is this accomplished by the
+            # constrained type in `datatypes.py`?
+            self.comm_session.emaid = payment_details_req.emaid
             self.comm_session.contract_cert_chain = payment_details_req.cert_chain
 
-            payment_details_res = PaymentDetailsRes(
-                response_code=ResponseCode.OK,
-                gen_challenge=get_random_bytes(16),
-                evse_timestamp=time.time(),
+            hash_data = self._get_contract_certificate_hash_data(
+                self.comm_session.contract_cert_chain
+            )
+            pem_certificate_chain = self._build_pem_certificate_chain(
+                self.comm_session.contract_cert_chain
             )
 
-            self.create_next_message(
-                Authorization,
-                payment_details_res,
-                Timeouts.V2G_SECC_SEQUENCE_TIMEOUT,
-                Namespace.ISO_V2_MSG_DEF,
+            authorization_result = (
+                await self.comm_session.evse_controller.is_authorized(
+                    id_token=payment_details_req.emaid,
+                    id_token_type=self.comm_session.selected_auth_option,
+                    certificate_chain=pem_certificate_chain,
+                    hash_data=hash_data,
+                )
             )
+
+            if authorization_result == AuthorizationStatus.ACCEPTED:
+
+                payment_details_res = PaymentDetailsRes(
+                    response_code=ResponseCode.OK,
+                    gen_challenge=get_random_bytes(16),
+                    evse_timestamp=time.time(),
+                )
+
+                self.create_next_message(
+                    Authorization,
+                    payment_details_res,
+                    Timeouts.V2G_SECC_SEQUENCE_TIMEOUT,
+                    Namespace.ISO_V2_MSG_DEF,
+                )
+            else:
+                # TODO: handle ONGOING case, and REJECTED more fully
```

Comment:
isnt this part quite straight forward? in case of Ongoing or Accepted we go to the Authorization state.
in case of rejected than we can answer with an Error code.

we also need to save the state of the authorization in the session here, I believe, because if the auth is successful in the PaymentDetailsReq, then we wont need to request it in the Authorization state and we can move forward. Unless, the ocpp client holds the authorization response and answer immediately if we do another request for the same eMAID.

if the answer "accepted" was not yet received, then we will keep requesting until the timeout (60s)

Reply:
Attached this to the existing https://github.com/SwitchEV/iso15118/issues/54